packages/odyssey-react/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Children, isValidElement, cloneElement, useMemo, CSSProperties, ComponentPropsWithRef, forwardRef } from "react";
+import type { ReactNode } from "react";
+import { nanoid } from 'nanoid';
+
+import styles from "./Icon.module.scss";
+
+export type Props = {
+  /**
+   * Title text used by screen readers
+   * @default the name of the icon
+   */
+  title?: string;
+
+  /**
+   * Id used to link title and svg elements, used by screen readers
+   * @default a randomly generated id
+   */
+  titleId?: string;
```

Comment:
I'm not sure we want to expose this as part of our API, consider that the <title> element is rendered as part of the SVG itself so there is little utility in making this configurable to outside parties. Instead, we should allow title text to be inserted as we do already via the `title` prop